packages/concerto-core/lib/decoratorextractor.js

```diff
@@ -227,30 +267,57 @@ class DecoratorExtractor {
         let vocabData = [];
         Object.keys(this.extractionDictionary).forEach(namespace => {
             const jsonData = this.extractionDictionary[namespace];
-            const patternToDetermineVocab = /^Term_/i;
             let dcsObjects = [];
             let vocabObject = {};
             jsonData.forEach(obj =>{
                 const decos = JSON.parse(obj.dcs);
                 const target = this.constructTarget(namespace, obj);
                 decos.forEach(dcs =>{
-                    if (dcs.name !== 'Term' && !patternToDetermineVocab.test(dcs.name)){
+                    const isVocab = this.isVocabDecorator(dcs.name);
+                    if (!isVocab && this.action !== DecoratorExtractor.Action.EXTRACT_VOCAB){
                         dcsObjects = this.parseNonVocabularyDecorators(dcsObjects, dcs, this.dcs_version, target);
                     }
-                    else {
+                    if (isVocab && this.action !== DecoratorExtractor.Action.EXTRACT_NON_VOCAB){
                         vocabObject = this.parseVocabularies(vocabObject, obj, dcs);
                     }
                 });
             });
-            decoratorData = this.transformNonVocabularyDecorators(dcsObjects, namespace, decoratorData);
-            vocabData = this.transformVocabularyDecorators(vocabObject, namespace, vocabData);
+            if(this.action !== DecoratorExtractor.Action.EXTRACT_VOCAB){
+                decoratorData = this.transformNonVocabularyDecorators(dcsObjects, namespace, decoratorData);
+            }
+            if(this.action !== DecoratorExtractor.Action.EXTRACT_NON_VOCAB){
+                vocabData = this.transformVocabularyDecorators(vocabObject, namespace, vocabData);
+            }
         });
         return {
             decoratorCommandSet: decoratorData,
             vocabularies: vocabData
         };
     }
-
+    /**
+     * Filter vocab or non-vocab decorators
+     * @param {Object} decorators - the collection of decorators
+     * @returns {Object} - the collection of filtered decorators
+     * @private
+     */
+    filterDecorators(decorators){
+        if(this.removeDecoratorsFromModel){
+            if (this.action === DecoratorExtractor.Action.EXTRACT_ALL){
+                return undefined;
```

Comment:
if removeDecoratorsFromModel is true then it should return undefined to remove all the decorators from the model

Reply:
Is there a disconnect between the `Action` and the logic then?

Reply:
The `action` parameter specifies what to extract from the model (vocab decorators, non-vocab decorators, or both). Meanwhile, `removeDecoratorsFromModel` flag determines if the decorators should be deleted or not from the model itself after extraction.

Reply:
This function then can be called `filterOutDecorators`. And then, I think you can just return early if `this.removeDecoratorsFromModel` is false with the decorators and avoid wrapping the whole function definition in an if statement.

Otherwise it looks like the code is looking like doing the opposite of what the function is aiming to do.

Reply:
Updated